cabal-install/src/Distribution/Client/CmdInstall.hs

```diff
@@ -649,7 +553,132 @@ installAction flags@NixStyleFlags{extraFlags = clientInstallFlags', ..} targetSt
       actionOnExe <- action v overwritePolicy <$> prepareExeInstall cfg
       traverse_ actionOnExe . Map.toList $ targetsMap buildCtx
 
--- | Treat all direct targets of install command as local packages: #8637 and later #7297, #8909, #7236.
+withProject
+  :: Verbosity
+  -> ProjectConfig
+  -> [String]
+  -> Bool
+  -> IO ([PackageSpecifier UnresolvedSourcePackage], [TargetSelector], ProjectConfig)
+withProject verbosity cliConfig targetStrings installLibs = do
+  -- First, we need to learn about what's available to be installed.
+  baseCtx <- establishProjectBaseContext reducedVerbosity cliConfig InstallCommand
+
+  (specs, selectors) <-
+    -- if every selector is already resolved as a packageid, return without further parsing.
+    if null unresolvedTargetStrings
+      then return (parsedSpecifiers, parsedTargets)
+      else do
+        (resolvedSpecifiers, resolvedTargets) <-
+          resolveTargetSelectorsInProjectBaseContext verbosity baseCtx targetStrings targetFilter
+        return (resolvedSpecifiers ++ parsedSpecifiers, resolvedTargets ++ parsedTargets)
+
+  -- Treat all direct targets of install command as local packages, see note in 'installAction'
+  let config =
+        addLocalConfigToPkgs (projectConfig baseCtx) $
+          concatMap (targetPkgNames $ localPackages baseCtx) selectors
```

Comment:
We no longer pass `specs` to `addLocalConfigToPkgs`, why so? Could you remind me what the `specs` here are?

Reply:
If a target string had to be resolved inside the project conterxt, then pkgSpecs will include the project packages turned into source distributions. We want to apply the local configuration only to the actual targets. This was basically what prompted me to clarify the behaviour we wanted. I'll add this as a note.